satellite/marketing/marketingweb/server_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package marketingweb_test
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/satellite/marketing"
+)
+
+func buildForm(isReferralOffer bool) url.Values {
+	form := url.Values{}
+	form.Add("Name", "May Credit")
+	form.Add("Description", "desc")
+	form.Add("ExpiresAt", "2019-06-27")
+	form.Add("InviteeCreditInCents", "50")
+	form.Add("InviteeCreditDurationDays", "50")
+	form.Add("RedeemableCap", "150")
+
+	if isReferralOffer {
+		form.Add("AwardCreditInCents", "50")
+		form.Add("AwardCreditDurationDays", "50")
+	}
+	return form
+}
+
+func buildResources(address, endpoint string, isReferralOffer bool) (url.URL, url.Values, error) {
+	URL, err := url.ParseRequestURI("http://" + address)
+	if err != nil {
+		fmt.Printf("err from buildResources : %v\n", err)
+		return *URL, url.Values{}, err
+	}
+	URL.Path = endpoint
+	form := buildForm(isReferralOffer)
+	URL.RawQuery = form.Encode()
+	return *URL, form, nil
+}
+
+func TestCreateOffer(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		validOffers := []marketing.NewOffer{
```

Comment:
Instead of making this of `marketing.NewOffer` here create a table of:

```
type CreateRequest struct {
    Path string
    Values url.Values
    ExpectedStatus http.StatusCode
}

requests := []CreateRequest{
    {
        Path: "/create/referral-offer,
        Values: url.Values{
            "Name", "May Credit",
            "Description", "desc",
            "ExpiresAt", "2019-06-27",
            "InviteeCreditInCents", "50",
            "InviteeCreditDurationDays", "50",
            "RedeemableCap", "150",
        },
        ExpectedStatus: http.StatusOK,
        RedirectedTo: "/",
    }
}
```

Reply:
`http.StatusCode` doesn't exist but `http.ResponseWriter` has a `StatusCode` field so I'm thinking you meant that or `http.StatusSeeOther` which you also can't use because it's not a type so wouldn't the right thing to do be to type-set `ExpectedStatus` as an int? 

The Path key in your `CreateRequest` struct literal has a string value that's not terminated by a closing double-quote.

In the `Values` key which is of type `map[string][]string` your formatting there looks wrong to me because your map literal has no key.

@egonelbre

Reply:
.........But I understand what you wan't me to do.

Reply:
https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go

Reply:
@VinozzZ can you elaborate please ?

Reply:
Ok I believe I know how to correct it.

Reply:
Yeah, my typos.

Reply:
No worries. I think you were just showing me the general idea. I can't expect you to take the time to spell it out for me every time. Thank you